client/src/js/components/bhDateInterval.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * @name bhDateInterval
+ * @description
+ * The `bhDateInterval` component provide a mean to select dates plage between
+ * two dates. The dates values returned are send to dates models given in
+ * date-from and date-to attributes
+ *
+ * @example
+ * ```html
+ * <bh-date-interval date-from="$MyCtrl.dateFrom" date-to="$MyCtrl.dateTo" >
+ * </bh-date-interval>
+ * ```
+ */
+angular.module('bhima.components').component('bhDateInterval', {
+  templateUrl : '/partials/templates/bhDateInterval.tmpl.html',
+  controller : bhDateInterval,
+  bindings : {
+    dateFrom : '=',
+    dateTo : '='
+  }
+});
+
+// dependencies injection
+bhDateInterval.$inject = ['DateService'];
+
+// controller definition
+function bhDateInterval(Dates) {
+  var vm = this;
+
+  vm.options = [
+    { translateKey : 'FORM.LABELS.TODAY', fn : day },
+    { translateKey : 'FORM.LABELS.THIS_WEEK', fn : week },
+    { translateKey : 'FORM.LABELS.THIS_MONTH', fn : month },
+    { translateKey : 'FORM.LABELS.THIS_YEAR', fn : year },
+    { translateKey : 'FORM.LABELS.OTHER', fn : other },
+  ];
+
+  // expose to the viewe
+  vm.search = search;
+  vm.clear  = clear;
+
+  // start up the modal
+  startup();
+
+  function search (selection) {
+    vm.selected = selection.translateKey;
+    selection.fn();
+  }
+
+  function day() {
+    vm.dateFrom = Dates.current.day();
+    vm.dateTo = new Date();
+  }
+
+  function week() {
+    vm.dateFrom = Dates.current.week();
+    vm.dateTo = new Date();
+  }
+
+  function month() {
+    vm.dateFrom = Dates.current.month();
+    vm.dateTo = new Date();
+  }
+
+  function year() {
+    vm.dateFrom = Dates.current.year();
+    vm.dateTo = Dates.current.year();
+    vm.dateTo.setMonth(11);
```

Comment:
This filter will look into the future.  Why not have it look from the beginning of this year to today?  That's what all the other filters do...  Are you expecting future values?

I would change this to `vm.dateTo = new Date()`.

If you need to actually look to the end of the year, I would use `momentjs` for clarity.  See the [endOf](http://momentjs.com/docs/#/manipulating/end-of/).